src/Build/Definition/Project.cs

```diff
@@ -1117,32 +1129,161 @@ public List<GlobResult> GetAllGlobs(string itemType)
             return GetAllGlobs(GetItemElementsByType(GetEvaluatedItemElements(), itemType));
         }
 
+        private struct CumulatedRemoveElementData
```

Comment:
Cumulative. As the items are scanned, that struct grows with every new piece of information on Remove operations. I'll rename it.